contracts/liquidity_hub/pool-manager/src/swap/perform_swap.rs

```diff
@@ -99,36 +103,60 @@ pub fn perform_swap(
     if offer_asset.denom == pools[0].denom {
         pair_info.assets[0].amount += offer_amount;
         pair_info.assets[1].amount -= swap_computation.return_amount;
+        pair_info.assets[1].amount -= swap_computation.protocol_fee_amount;
+        pair_info.assets[1].amount -= swap_computation.burn_fee_amount;
     } else {
         pair_info.assets[1].amount += offer_amount;
         pair_info.assets[0].amount -= swap_computation.return_amount;
+        pair_info.assets[0].amount -= swap_computation.protocol_fee_amount;
+        pair_info.assets[0].amount -= swap_computation.burn_fee_amount;
     }
```

Comment:
don't we need to take into account the osmosis_fee here too?

Reply:
We want to, coming on a refactoring